docs/source/DeveloperGuide/AddingNewComponents/DevelopingCommands/index.rst

```diff
@@ -31,9 +35,33 @@ The instructions below must be applied as close as possible although some variat
 
 which will set the interpreter directive to the python on the environment.
   
-**2.** The next is the documentation line which is describing the command. This same documentation line will be used also the command help information available with the *-h* command switch.   
+**2.** The next is the documentation line which is describing the command. This same documentation line will be used also the command help information available with the *-h* command switch.
+
+**3.** The majority of the code should be contained with a function, often called ``main`` though this is not required. This function should be wrapped with the ``@DiracScript()`` decorator to allow the DIRAC plugin mechanism to override the script with the function from the highest priority extension.
+
+.. code-block:: python
+
+   #Import the required DIRAC modules
+   from DIRAC.Core.Utilities.DIRACScript import DIRACScript
+   from DIRAC.Interfaces.API.DIRAC import DIRAC
+   from DIRAC import gLogger
+
+   @DiracScript()
+   def main()
+     # Do stuff
+
+   if __name__ == "__main__":
+     main()
+
+**4.** Next the function must be registered as a ``console_scripts`` ``entrypoint`` in the ``setuptools`` metadata (`more details <https://setuptools.readthedocs.io/en/latest/userguide/entry_point.html>`_). This is done by adding a line to the ``console_scripts`` list like below, where the first string is the name for the script you want to create, the left hand side of ``:`` is the module that contains your function and the right hand side is the object you want to invoke (e.g. a function).
```

Comment:
explicitly mention `setup.cfg`. I don't see it here

Reply:
That was stupid, fixed in ef26bd1